src/core/pointcloud/qgseptdecoder.cpp

```diff
@@ -517,62 +529,83 @@ QgsPointCloudBlock *__decompressLaz( FileType &file, const QgsPointCloudAttribut
     return block.release();
   }
 
-  lazperf::las::point10 p;
+  lazperf::las::point10 p10;
   lazperf::las::gpstime gps;
   lazperf::las::rgb rgb;
+  lazperf::las::point14 p14;
 
   for ( size_t i = 0 ; i < count ; i ++ )
   {
     f.readPoint( buf ); // read the point out
-    p.unpack( buf );
-    gps.unpack( buf + sizeof( lazperf::las::point10 ) );
-    rgb.unpack( buf + sizeof( lazperf::las::point10 ) + sizeof( lazperf::las::gpstime ) );
 
+    // LAS 1.2 file support
+    if ( lasPointFormat == 3 )   // base + gps time + rgb
```

Comment:
Although our untwine produced only *format 3* laz files, the previous code would also correctly read existing ept files with formats *0 and 1* generated by other software. Maybe even format *2* but with wrong values for the gpstime.
I think it would be nice to handle those formats too, by simply unpacking gps/rgb depending on the format.

Reply:
Should be addressed in 056ae34